std/experimental/allocator/building_blocks/quantizer.d

```diff
@@ -75,6 +75,7 @@ struct Quantizer(ParentAllocator, alias roundingFunction)
     */
     void[] allocate(size_t n)
     {
+        if (n == 0) return null;
```

Comment:
Is this necessary? If parent.allocate returns `null`, then that should automatically be propagated by the ternary expression on line 80.

Reply:
This is might be necessary because `parent.allocate` might not return `null`, which is exactly what is happening with `Region.allocate`. Please see #5830.
Another reason would be to avoid calling `parent.allocate`, since this should return `null`.

What do you think?

Reply:
As far as I can tell from [the documentation](https://dlang.org/phobos/std_experimental_allocator_building_blocks.html), passing 0 to allocate should always return `null`, and if it doesn't, that's a bug and should be correct "upstream".